backend/drone.py

```diff
@@ -20,9 +20,11 @@ def __del__(self):
         """
         Destructor to ensure the vehicle connection is closed when the DroneController object is deleted.
         """
-        if hasattr(self, 'vehicle'):
-            self.vehicle.close()
-    
+        try:
```

Comment:
try and finally is not needed as it is a simple del function which has only one purpose.